### ISSUE TITLE
What task queue should the Promise be queued on?

### ISSUE THREAD

**Issue by yoavweiss** [WICG/ua-client-hints#77]
In [getEntropyValues](https://wicg.github.io/ua-client-hints/#getHighEntropyValues), we're queueing a Promise to be resolved, but it's unclear on what task queue that should happen. We need to figure out.

/cc @domenic @bzbarsky

**Comment by domenic** [comment]
In Chromium, the plan is apparently to not go in parallel, and just to resolve the promise synchronously. (But promise reactions still run synchronously, as guaranteed by the promise machinery.)

However, presumably the reason this is a promise at all is to allow future flexibility in doing some in-parallel work before resolving. (E.g., prompting the user if the [privacy budget](https://github.com/bslassey/privacy-budget) has been exceeded. Note: I have no idea if this is the actual plan; this is just an example.)

So this raises an interesting spec-writing question.

- Do we spec and implement sync resolution now, then maybe change it later once an implementer is interested in doing so?

- Do we spec async resolution with some task source (see discussion in https://github.com/whatwg/html/issues/5329#issuecomment-594177261 about which one), and implement sync resolution?

- Do we spec async resolution, and also implement it, even though it's not necessary in the implementation? This amounts to adding an artificial delay.

- Do we spec async resolution with a new task source just for this spec? Then I _think_ a sync-resolution implementation (such as Chromium's) would be conformant, if we said that the implementation prioritized this task source at the highest-possible priority.

In general, any changes in resolution timing have minor compat risks, similar to the ones discussed in https://github.com/whatwg/html/issues/5329#issuecomment-594177261. You could theoretically write a test that distinguishes between sync resolution and async resolution (unless we had a dedicated task source... I think). But they're pretty minor, so I think this is more a matter of how best to write the spec to give implementations flexibility and to communicate the intent...

**Comment by bzbarsky** [comment]
> You could theoretically write a test that distinguishes between sync resolution and async resolution (unless we had a dedicated task source... I think)

Task source doesn't matter:
```
  Promise.resolve().then(() => console.log(1));
  myMethod().then(() => console.log(2));
  Promise.resolve().then(() => console.log(3));
```
will log `1,2,3` if the resolution is sync and `1,3,2` if it's async, independent of task source.

**Comment by yoavweiss** [comment]
> In Chromium, the plan is apparently to not go in parallel, and just to resolve the promise synchronously.

To clarify, this is how it's currently implemented, but can be modified, if the conclusion of this issue is that it creates a future compat risk.

**Comment by yoavweiss** [comment]
After thinking about this some more, the [permission task queue](https://w3c.github.io/permissions/#permission-task-source) seems like the most appropriate one.
I'm [aligning](https://chromium-review.googlesource.com/c/chromium/src/+/2166103/) the Chromium implementation on that. Let me know if you have any objections.
